### PR TITLE
feat: adapt volcano apis to k8s 1.31

### PR DESCRIPTION
**Pull request by vie-serendipity**
adpat volcano apis to 1.31 kubernetes
Ref: https://github.com/volcano-sh/volcano/issues/3765

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,71 +1,72 @@
 module volcano.sh/apis
 
-go 1.22.0
+go 1.23.0
```
[the remark this answers is not in the document]
Comment (vie-serendipity) [reply go.mod:3]: It' not needed, I messed up and used the master version.

**Review by william-wang — APPROVED** [review]
/approve

**Comment by dongjiang1989** [comment]
@william-wang tag apis to v1.11.0.
I want volcano adapts to the k8s v1.31

**Comment by Monokaix** [comment]
> @william-wang tag apis to v1.11.0. I want volcano adapts to the k8s v1.31

Hi, we will release next version about one month later, can you just wait the new version being released.

**Comment by dongjiang1989** [comment]
> > @william-wang tag apis to v1.11.0. I want volcano adapts to the k8s v1.31
> 
> Hi, we will release next version about one month later, can you just wait the new version being released.

Thanks. @Monokaix  I got it 
Can add https://github.com/volcano-sh/apis/pull/139 PR to apis v1.11.0?

**Pull request merged**